### ISSUE TITLE
Component: Editor isn't functional within dialogs

### ISSUE THREAD

**Issue by aaron-cardenas** [primefaces/primeng#15324]
### Describe the bug

Since the EditorModule fix doesn't support SSR in v17.13.0, the editor isn't functional within dialogs. The same issue persists in 17.14.x. Reverting to the 17.12.x fix leaves you with a choice between this functionality and SSR compatibility.
https://github.com/primefaces/primeng/issues/8817

### Environment

Angular version 17.3.5
PrimeNG version 17.13.0 - 17.14.x
Node version 18.19.1

### Angular version

17.3.5

### PrimeNG version

17.13.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.19.1

### Browser(s)

All

### Steps to reproduce the behavior

Open a dialog panel (p-dialog) containing an Editor, it dosent work and don't have styles.

**Comment by armingdev** [comment]
I hope this will be fixed soon. I'm blocked on the project because of this issue. #14931

**Comment by nejat-njonjo** [comment]
You should put the editor inside if block, so it gets initialized on when the visible state changes, like below. I use it like this and it works jus fine.

```html
<p-dialog
  header="some header"
  [modal]="true"
  [(visible)]="visible"
  [style]="{ width: '50rem' }"
  [breakpoints]="{ '1199px': '75vw', '575px': '90vw' }"
  [maximizable]="true"
>
    @if (visible) {
     <p-editor></p-editor>
    }
</p-dialog>
```

**Comment by mertsincan** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,
